### PR TITLE
reduce cyclomatic complexity of Reconcile function 

### PR DESCRIPTION
**Pull request by barney-s**
### Change description

Reduce cyclomatic complexity of Reconcile() function:

* Seperate code that gets a new planCR into its own function
* Separate code that updates plan status into its own function
* Refactor a nested if check

### CODE REVIEW

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 1]
```diff
@@ -77,41 +77,37 @@ var contextGVK schema.GroupVersionKind = schema.GroupVersionKind{
...
+		nn := types.NamespacedName{Namespace: plancr.Namespace, Name: plancr.Name}
+		err := r.Client.Get(ctx, nn, plancr)
+		if err != nil {
+			return err
 		}
+		plancr.Status = *newStatus.DeepCopy()
+		return r.Client.Status().Update(ctx, plancr)
+	})
+	if err != nil {
+		logger.Error(err, "unable to update Plan status")
```
Comment (cheftako) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:93]: Please include the error in the log

Comment (cheftako) [reply experiments/compositions/composition/internal/controller/expander_reconciler.go:93]: Ideally the GVK/identifier as well.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 2]
```diff
@@ -126,21 +122,59 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 				Stages: map[string]*compositionv1alpha1.StageStatus{},
 			},
 		}
-		if err := ctrl.SetControllerReference(&inputcr, &plancr, r.Scheme); err != nil {
+		if err := ctrl.SetControllerReference(inputcr, &plancr, r.Scheme); err != nil {
 			logger.Error(err, "Unable to set controller reference for Plan Object")
```
Comment (cheftako) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:126]: Lets log the error.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 3]
```diff
@@ -126,21 +122,59 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
...
+func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger = logger.WithName(r.Composition.Name).WithName(r.InputGVK.Group)
+
+	logger.Info("Got Input API for expansion", "request", req)
+
+	inputcr := unstructured.Unstructured{}
+	inputcr.SetGroupVersionKind(r.InputGVK)
+	if err := r.Get(ctx, req.NamespacedName, &inputcr); err != nil {
+		logger.Error(err, "unable to fetch Input API Object")
```
Comment (cheftako) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:152]: We should indicate which API object we could not fetch.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go (hunk 1 above)
Comment (barney-s) [reply experiments/compositions/composition/internal/controller/expander_reconciler.go:93]: resolved.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go (hunk 2 above)
Comment (barney-s) [reply experiments/compositions/composition/internal/controller/expander_reconciler.go:126]: logger.Error is passed the err object. Iam assuming that will be logged. Added the plan name and namespace.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go (hunk 3 above)
Comment (barney-s) [reply experiments/compositions/composition/internal/controller/expander_reconciler.go:152]: done.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 4]
```diff
@@ -77,41 +77,37 @@ var contextGVK schema.GroupVersionKind = schema.GroupVersionKind{
...
 	}
+}
 
-	// Associate a plan object with this input CR
+func (r *ExpanderReconciler) getPlanForInputCR(ctx context.Context, inputcr *unstructured.Unstructured) (types.NamespacedName,
+	*compositionv1alpha1.Plan, error) {
 	var plancr compositionv1alpha1.Plan
-	planNN := types.NamespacedName{Name: r.InputGVR.Resource + "-" + inputcr.GetName(), Namespace: inputcr.GetNamespace()}
+	logger := log.FromContext(ctx)
+	logger = logger.WithName(r.Composition.Name).WithName(r.InputGVK.Group).WithName(inputcr.GetName())
```
Comment (justinsb) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:101]: Q: do the names replace each other?

**Review by justinsb — APPROVED** [review]

File: experiments/compositions/composition/internal/controller/expander_reconciler.go (hunk 4 above)
Comment (barney-s) [reply experiments/compositions/composition/internal/controller/expander_reconciler.go:101]: I have seen it append:

`compositionName.Group.inputcrName : log here `

**Pull request merged**